crates/sdl-encoder/src/field_value.rs

```diff
@@ -0,0 +1,120 @@
+use std::fmt::{self, Display};
+
+/// Convenience Field Value implementation used when creating a Field.
+/// Can be a `Type`, a `NonNull` or a `List`.
+///
+/// This enum is resposible for encoding creating values such as `String!`, `[[[[String]!]!]!]!`, etc.
+///
+/// ### Example
+/// ```rust
+/// use sdl_encoder::{FieldValue};
+///
+/// let field_ty = FieldValue::Type {
+///     ty: "String".to_string(),
+/// };
+///
+/// let list = FieldValue::List {
+///     ty: Box::new(field_ty),
+/// };
+///
+/// let non_null = FieldValue::NonNull { ty: Box::new(list) };
+///
+/// assert_eq!(non_null.to_string(), "[String]!");
+/// ```
+#[derive(Debug, PartialEq, Clone)]
+pub enum FieldValue {
+    /// The non-null field type.
+    NonNull {
+        /// Null inner type.
+        ty: Box<FieldValue>,
```

Comment:
Maybe `ofType`?

Reply:
It's more clear to use `ty` as other encoders do.

Reply:
It is not just a type though, it is a `NonNull` or `List` of a specific type.